### PR TITLE
Ng2 SettingsComponent karma spec

### PR DESCRIPTION
**Pull request by mohamed2a**
- fix PasswordComponent and HealthComponent spec files
- add SettingsComponent spec file

### CODE REVIEW

File: generators/client/templates/angular/src/test/javascript/spec/app/account/settings/_settings.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        it('should send the current identity upon save', function () {
+            //GIVEN
+            let accountValues = {
+                firstName: "John",
+                lastName: "Doe",
+
+                activated: true,
+                email: "john.doe@mail.com",
+                langKey: "en", // ici
+                login: "john"
```
Comment (ramzimaalej) [inline generators/client/templates/angular/src/test/javascript/spec/app/account/settings/_settings.component.spec.ts]: If social is enabled, we should send back the imageUrl as well.

**Comment by deepu105** [comment]
I dont know why the protractor fails for one of the sass + noi18n test. can you run it locally and let me know if it passes https://travis-ci.org/jhipster/generator-jhipster/jobs/194764197

File: generators/client/templates/angular/src/test/javascript/spec/app/account/settings/_settings.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
+import {ComponentFixture, TestBed, async, inject} from '@angular/core/testing';
+import {MockBackend} from '@angular/http/testing';
+import {Http, BaseRequestOptions} from '@angular/http';
+import {Observable} from 'rxjs/Rx';
+import {<%=jhiPrefixCapitalized%>LanguageService} from 'ng-jhipster';
+import {Principal, AccountService, <%=jhiPrefixCapitalized%>LanguageHelper} from '../../../../../../main/webapp/app/shared';
```
Comment (pascalgrimaud) [inline generators/client/templates/angular/src/test/javascript/spec/app/account/settings/_settings.component.spec.ts]: I think `<%=jhiPrefixCapitalized%>LanguageHelper` is only need when enableTranslation

File: generators/client/templates/angular/src/test/javascript/spec/app/account/settings/_settings.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+                            return new Http(backendInstance, defaultOptions);
+                        },
+                        deps: [MockBackend, BaseRequestOptions]
+                    },
+                    {
+                        provide: <%=jhiPrefixCapitalized%>LanguageService,
+                        useClass: MockLanguageService
+                    },
+                    {
+                        provide: <%=jhiPrefixCapitalized%>LanguageHelper,
```
Comment (pascalgrimaud) [inline generators/client/templates/angular/src/test/javascript/spec/app/account/settings/_settings.component.spec.ts]: same comment:
I think `<%=jhiPrefixCapitalized%>LanguageHelper` is only need when enableTranslation

**Pull request merged**